scipy/optimize/optimize.py

```diff
@@ -2397,20 +2397,81 @@ def bracket(func, xa=0.0, xb=1.0, args=(), grow_limit=110.0, maxiter=1000):
     return xa, xb, xc, fa, fb, fc, funcalls
 
 
-def _linesearch_powell(func, p, xi, tol=1e-3):
+def _m_min(a, b):
+    """ Return min(a, b) """
+    if a is None:
```

Comment:
The following might be preferable, especially if it is more common for both `a` and `b` to have values.
```
    try:
        return min(a, b)
    except ValueError:
        return a or b
```

Reply:
Done.